src/booleans.js

```diff
@@ -1,61 +1,74 @@
 function negate(a) {
-  // your code here
+  return !a;
 };
 
 function both(a, b) {
-  // your code here
+  return a && b;
 };
 
 function either(a, b) {
-  // your code here
+  return a || b;
 };
 
 function none(a, b) {
-  // your code here
+  return !(a || b);
 };
 
 function one(a, b) {
-  // your code here
+  return !(a===b);
 };
 
 function truthiness(a) {
-  // your code here
+  if (a) {
+    return true;
+  }
+  else {
+    return false;
+  }
 };
 
 function isEqual(a, b) {
-  // your code here
+  return a===b;
 };
 
 function isGreaterThan(a, b) {
-  // your code here
+  return a > b;
 };
 
 function isLessThanOrEqualTo(a, b) {
-  // your code here
+  return a <= b;
 };
 
 function isOdd(a) {
-  // your code here
+  return a % 2 === 1;
 };
 
 function isEven(a) {
-  // your code here
+  return 1+a % 2 === 1;
 };
 
 function isSquare(a) {
-  // your code here
+  return Math.sqrt(a) % 1 === 0 && a >= 0;
 };
 
 function startsWith(char, string) {
-  // your code here
+  return char === string.slice(0, 1);
 };
 
 function containsVowels(string) {
-  // your code here
-};
+  let vowelsCount = 0
+  const str = string.toLowerCase();
+  const vowels = ["a", "e", "i", "o", "u"];
+  for (let char of str) {
+    if (vowels.includes(char)){
+      vowelsCount++;
+    };
+  };
+  return vowelsCount > 0;
```

Comment:
So this feels like a good chance to introduce regex to you!
This is one of those things that you'll end up using at some point, its a very versatile and powerful way of matching patterns in strings very quickly. They are written in a format like `/[aeiou]/`, and once you get the hang of them they are really useful

What this example means is: match any part of a given string with any character between the square brackets. In this case it will only match these exact characters in lower case, but a small modifier of the `i` flag at the end (making it `/[aeiou]/i`) makes it case insensitive, so will match both cases of anything in the brackets.
These are used with the `.match` js function, like this:
```
someString.match(/[aeiou]/i)
```
this will return either true if it finds a match, or null if not. So for this function we could use 
```
  return string.match(/[aeiou]/i);
```
and it would return `true` if the string contained a lower or upper case vowel, and `null` if not. But since we only want a `true` or `false` value to be returned, we can use that double negation (!!) method again to convert the `null` value to false:
```
  return !!string.match(/[aeiou]/i);
```
Way more concise.